### DISCUSSION TITLE
Using pg-promise with an existing pg client / connection

### DISCUSSION THREAD

**Discussion by innermatrix** [vitaly-t/pg-promise#876] [Q&A] [answered]
I am using `pg-promise` throughout my code. However, in one area I am working with a 3rd party package that calls my code and passes it a pg `Client` object. I am hoping that I can keep using `pg-promise` inside this callback. How can I take an existing `Client` object and imbue it with `pg-promise` calls like `oneOrNone` etc? Thanks.

**Comment by vitaly-t** [comment, marked as answer]
This library is built mainly around a connection pool, not individual/separate connections.

The only thing this library has in terms of interconnectivity is property [db.$pool](https://vitaly-t.github.io/pg-promise/Database.html#$pool), which exposes the underlying pool object. There is nothing else. It cannot wrap itself around an external `Client` interface, I'm afraid.

In theory, it should be easy to implement, perhaps by extending method [connect]() and adding option `client`, it would just need some work ;)
